packages/schematics/src/app/rules.ts

```diff
@@ -0,0 +1,207 @@
+import { JsonAstObject, JsonParseMode, normalize, parseJsonAst } from '@angular-devkit/core';
+import {
+  apply,
+  branchAndMerge,
+  mergeWith,
+  move,
+  Rule,
+  SchematicContext,
+  SchematicsException,
+  template,
+  Tree,
+  url,
+} from '@angular-devkit/schematics';
+import { NodePackageInstallTask } from '@angular-devkit/schematics/tasks';
+import { getWorkspace, updateWorkspace } from '@schematics/angular/utility/config';
+import {
+  addPackageJsonDependency,
+  NodeDependency,
+  NodeDependencyType,
+} from '@schematics/angular/utility/dependencies';
+import {
+  appendPropertyInAstObject,
+  findPropertyInAstObject,
+  insertPropertyInAstObjectInOrder,
+} from '@schematics/angular/utility/json-utils';
+import { getProject } from '@schematics/angular/utility/project';
+import { getProjectTargets } from '@schematics/angular/utility/project-targets';
+import { of } from 'rxjs';
+import { concatMap, map } from 'rxjs/operators';
+
+import { getLatestNodeVersion, NpmRegistryPackage } from './npmjs-utils';
+import { Schema } from './schema';
+
+export function validateOptions(options: Schema): Rule {
+  return (tree: Tree, context: SchematicContext) => {
+    if (!options.project || options.project === '') {
+      throw new SchematicsException('You need to enter project name');
+    }
+
+    const workspace = getWorkspace(tree);
+    const project = getProject(workspace, options.project);
+
+    if (!project) {
+      throw new SchematicsException(`Could not find project: '${options.project}'`);
+    }
+
+    return tree;
+  };
+}
+
+export function addPackageJsonDependencies(): Rule {
```

Comment:
I think this code already exists in `import { addPackageJsonDependency, NodeDependency } from '@schematics/angular/utility/dependencies';` Any reason we did not use this? We can probably remove all this code in this method.

Reply:
I'm using it under the hood. The reason why I wrap it is because I need to get the latest version. Angular has it "hard coded", I guess the update the file using their build scripts. But my function calls the `addPackageJsonDependency` once I got the version.